src/EntityFramework/Core/EntityClient/DbParameterCollectionHelper.cs

```diff
@@ -292,6 +292,50 @@ private static int IndexOf(IEnumerable items, string parameterName)
             return -1;
         }
 
+        /// <inhertidoc />
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1604:ElementDocumentationMustHaveSummary")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1611:ElementParametersMustBeDocumented")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1615:ElementReturnValueMustBeDocumented")]
+        [SuppressMessage("Microsoft.Usage", "CA2201:DoNotRaiseReservedExceptionTypes")]
+        public DbParameter GetParameterByValue(string value)
+        {
+            var index = IndexOfParameterValue(InnerList, value);
+            if (index < 0)
+            {
+                return null;
+            }
+
+            return InnerList[index];
+        }
+
+        private static int IndexOfParameterValue(IEnumerable items, string parameterValue)
```

Comment:
@bengutt thanks to you for contributing this fix! 

I will go ahead and provide some additional feedback on this PR although it is already merged. Depending on the timing we can take an additional PR from you or do some simplifications on top of what you did.